File: server/api/parser_api.py
from server.handlers.events import *


async def parser_api(message: dict, websocket: WebSocket) -> None:
    try:
        print(message)
        message_type = message['messageType']
        message_body = message['message']
        now_time = datetime.now()

        if message_type == MessageType.SUBSCRIBE_MARKET_DATA:
            await on_subscribe_market_data(websocket, message_body)

        elif message_type == MessageType.UNSUBSCRIBE_MARKET_DATA:
            await on_unsubscribe_market_data(websocket)

        elif message_type == MessageType.PLACE_ORDER:
            await on_place_order(websocket, message_body, now_time)

        elif message_type == MessageType.CANCEL_ORDER:
            await on_cancel_order(websocket, message_body, now_time)

        else:
            await on_unknown_event(websocket)

    except (KeyError, ValueError) as error:
        await websocket.send_json(ErrorInfo(reason=f'Ошибка в формате сообщения: {error}').to_json())

File: client/src/api/parser_api.js
import MarketDataUpdate from "./handlers/MarketDataUpdate";

const [
        SUBSCRIBE_MARKET_DATA,
        UNSUBSCRIBE_MARKET_DATA,
        PLACE_ORDER,
        CANCEL_ORDER,
        SUCCESS_INFO,
        ERROR_INFO,
        EXECUTION_REPORT,
        UPDATE_MARKET_DATA] = [1, 2, 3, 4, 5, 6, 7, 8]

export const parserApi = (message) => {
    switch (message['messageType']) {
        case SUBSCRIBE_MARKET_DATA:
            break
        case UNSUBSCRIBE_MARKET_DATA:
            break
        case PLACE_ORDER:
            break
        case CANCEL_ORDER:
            break
        case SUCCESS_INFO:
            break
        case ERROR_INFO:
            break
        case EXECUTION_REPORT:
            break
        case UPDATE_MARKET_DATA:
            MarketDataUpdate(message['message'])
            break
    }
}

export default parserApi;